### ISSUE TITLE
Bug: LTTP can create an extra TR small key, modifying the itempool later than it should

### ISSUE THREAD

**Issue by alwaysintreble** [ArchipelagoMW/Archipelago#2311]
### What happened?

If it doesn't sufficiently [satisfy certain conditions](https://github.com/ArchipelagoMW/Archipelago/blob/6befc917730a92380e894eb5235f5a599be73dfd/worlds/alttp/Rules.py#L956-L960) LTTP will [create an extra TR small key](https://github.com/ArchipelagoMW/Archipelago/blob/6befc917730a92380e894eb5235f5a599be73dfd/worlds/alttp/Rules.py#L860-L869), replacing a junk item from the itempool.

### What were the expected results?

Worlds shouldn't create or take from the multiworld itempool after the `create_items` step #1460 @espeon65536 

### Software

Local generation

**Comment by Exempt-Medic** [comment]
@Berserker66

**Comment by Berserker66** [comment]
Yeah, that's espeon's code. Unfortunately it's  not going to be easy to check for the conditions of  the key being required ahead of  time, as it needs a working all state.
